Doc 304 pt 3 - Finishing Document Store import (#160)
Macrometacorp/docs: docs/collections/documents/document-store-indexes.md

---
title: Document Store Indexes
sidebar_position: 50
---

> Indexes allow fast access to documents, provided the indexed attributes are used in a query, they speed up searches. For more informatin about indexes, refer to [Indexing](../indexing/index.md).

Reviewer: ```suggestion
Indexes enable you to create searchable attributes to speed up your queries. For more information about indexes, refer to [Indexing](../indexing/index.md).
```

## Document Store Indexes

You can create the following types of index for a document collection:

- [Geo Index](../indexing/geo-indexes.md)
- [Fulltext Index](../indexing/fulltext-indexes.md)
- [Persistent Index](../indexing/persistent-indexes.md)
- [TTL Index](../indexing/ttl-indexes.md)

Refer to each index type for more information about the index and for instructions on how to create and use it.

## View Indexes

View active indexes for a collection on the Indexes tab. The section explains what each element of the tab means.

![Document Store Data Tab](/img/collections/doc-store-indexes.png)

> - **ID -** This is a primary key for the indexes - no two documents can have the same ID.

Reviewer: ```suggestion
- **ID -** This is a unique primary key for the indexes.
```

- **Type -** The index type.

> - **Unique -** If the index is declared unique, then no two documents are allowed to have the same set of attribute values. The **true** option is the default value for primary keys/indexes while **false** is specified for all other keys/indexes.

Reviewer: ```suggestion
- **Unique -** If `true`, then no two documents are allowed to have the same set of attribute values. Default is `true` for primary keys and indexes, and default is `false` is for all other keys and indexes.
```

> - **Sparse -** Indicates whether the index is sparse or not. If the index is declared sparse, then a document will be excluded from the index and no uniqueness checks are performed if any index attribute value is not set or has a value of null.

Reviewer: ```suggestion
- **Sparse -** If `true`, then a document is excluded from the index. If any index value is not set or has a null value, GDN does not perform uniqueness checks.
```

Should confirm that the options are `true` or `false`

> - **Deduplicate -** Controls whether inserting duplicate index values from the same document into a unique array index will lead to a unique constraint error or not. Default is **true**, meaning that only a single instance of each non-unique index value will be inserted into the index per document. If **false**, then each instance of a non-unique index value will be inserted into the index per document.

Reviewer: ```suggestion
- **Deduplicate -** If `true` (default), GDN only inserts each non-unique index value once per document. Attempting to insert duplicate values results in an error. If `false`, GDN inserts each instance of the value into the index per document.
```

> - **Extras -** Extra conditions of the index definition, such as minimum length for fulltext index.

Reviewer: ```suggestion
- **Extras -** Extra conditions of the index definition, such as minimum length for full-text index.
```
